graphs/util/gpml_util_interface.py

```diff
@@ -0,0 +1,296 @@
+import graphspace_interface as interface
+import networkx as nx
+import json
+import xml.etree.ElementTree as ET
+
+
+# Current supported GPML format
+GPML = '{http://pathvisio.org/GPML/2013a}'
+
+
+def parse_gpml(graph_gpml, title):
+    """ Parse the GPML file
+    """
+
+    # Create a directed NetworkX graph to parse through
+    G = nx.DiGraph(directed=True)
+    # Use xml parser to parse the GPML file
+    graph_gpml = ET.fromstring(graph_gpml)
+    # Keep the overall count of nodes and create unique ids for
+    # datanodes, shape, labels.
+    # IMPORTANT!!!
+    count = 0
+
+    # Parsing the GPML file for Shape elements
+    G, count = parse_shapes(G, count, graph_gpml)
+
+    # Find all the DataNodes in the GPML file which is used to draw various
+    # the various nodes of the graph
+    G, count = parse_datanodes(G, count, graph_gpml)
+    # print G.node
+    # Parsing the GPML file for Shape elements
+    G, count = parse_labels(G, count, graph_gpml)
+
+
+    # This is a crude version of getting interactions (edges) working
+    # Look for all the interactions
+    interactions =  graph_gpml.findall(GPML+'Interaction')
+    G, count = parse_interactions(G, count, interactions)
+    layout = set_gpml_layout(G)
+
+
+    # To add nodes to the group we parse through the group
+    # elements of the GPML file.
+    groups =  graph_gpml.findall(GPML+'Group')
+    G = parse_groups(G, groups)
+
+    
+    metadata = {'description':'','title':  graph_gpml.attrib['Name'] or "GPML",'tags':[]}
+    title = title or metadata['title']
+    parse_json = interface.convertNXToDict(G, metadata=metadata)
+
+    return parse_json, layout, title
+
+
+def parse_shapes(G, count,  graph_gpml):
+    # Create a list of details of all the shape
+    shapes =  graph_gpml.findall(GPML+'Shape')
+    shape_details = []
+    for shape in shapes:
+        temp_node = {}
+        temp_node.update(shape.attrib)
+        for more_details in shape:
+            temp_node.update(more_details.attrib)
+        shape_details.append(temp_node)
+    # Add shape elemets to the JSON structure parse_json
+    for node in shape_details:
+        temp_node = {'data': {}}
```

Comment:
Use GraphId as the unique id for this shape. Do not use count anymore. Of course, you should ensure that all GraphIds over the graph are unique, just out of caution.